chapter_12_dependency_injection.asciidoc

```diff
@@ -631,6 +642,18 @@ Your MessageBus is a global object and putting queue on it – and clearing it
 out when handle is called is both hard to follow and 100% not thread safe.
 ////
 
+////
+ (EJ1) re: above comment, there's just *A BUG YOU MUST FIX*. It should be:
+
+def handle(self, message):
+    queue = [message]
```

Comment:
i'd have to reintroduce `queue` as an argument to `self.hande_event()` and `self.handle_command()` which is a shame but at least i won't have the obvious threading gotcha i guess. nice one.

Reply:
Ah, I see what you were going for now.  The pseudo-snippet is the simplest+most-transparent way, but I suppose you could create yet another level of indirection, when you register the handler:
```
class MessageBus:
    def register_handler(self, event_type, hander, *args, *kwargs):
        f = partial(handler, queue=[], *args, **kwargs)
        self.handlers[event_type] = f
```
Or do threadlocal storage, but that would be event more obtuse.

Reply:
Nvm, that `register_handler` snippet has the same problem with threads.  Basically you'd need to introduce another abstraction that wraps the handler and manages the queue lifecycle for you.